### PR TITLE
Feature/base macro passthrough

### PR DESCRIPTION
**Pull request by emielver**
## Description & motivation
Add base macro rebase and passthrough fields for mobile. Also unified int test data

## Checklist
- [x] I have verified that these changes work locally
- [x] I have added tests & descriptions to my models (and macros if applicable)

### CODE REVIEW

File: dbt_project.yml [hunk 1]
```diff
@@ -29,20 +29,21 @@ vars:
...
-    snowplow__application_context: "{{ source('atomic','com_snowplowanalytics_mobile_application_1') }}"
-    snowplow__screen_context: "{{ source('atomic','com_snowplowanalytics_mobile_screen_1') }}"
-    snowplow__app_errors_table: "{{ source('atomic','com_snowplowanalytics_snowplow_application_error_1') }}"
-    snowplow__screen_view_events: "{{ source('atomic','com_snowplowanalytics_mobile_screen_view_1') }}"
+    snowplow__session_context: "com_snowplowanalytics_snowplow_client_session_1"
+    snowplow__mobile_context: "com_snowplowanalytics_snowplow_mobile_context_1"
+    snowplow__geolocation_context: "com_snowplowanalytics_snowplow_geolocation_context_1"
+    snowplow__application_context: "com_snowplowanalytics_mobile_application_1"
+    snowplow__screen_context: "com_snowplowanalytics_mobile_screen_1"
+    snowplow__app_errors_context: "com_snowplowanalytics_snowplow_application_error_1"
```
Comment (agnessnowplow) [inline dbt_project.yml:37]: Why the change here? If it is an sde it is better to rename it to `_events`. FYI I started to follow this convention in the unified model: 
image

Comment (emielver) [reply dbt_project.yml]: Yep, you're right!

File: dbt_project.yml [hunk 2]
```diff
@@ -29,20 +29,21 @@ vars:
...
-    snowplow__application_context: "{{ source('atomic','com_snowplowanalytics_mobile_application_1') }}"
-    snowplow__screen_context: "{{ source('atomic','com_snowplowanalytics_mobile_screen_1') }}"
-    snowplow__app_errors_table: "{{ source('atomic','com_snowplowanalytics_snowplow_application_error_1') }}"
-    snowplow__screen_view_events: "{{ source('atomic','com_snowplowanalytics_mobile_screen_view_1') }}"
+    snowplow__session_context: "com_snowplowanalytics_snowplow_client_session_1"
+    snowplow__mobile_context: "com_snowplowanalytics_snowplow_mobile_context_1"
+    snowplow__geolocation_context: "com_snowplowanalytics_snowplow_geolocation_context_1"
+    snowplow__application_context: "com_snowplowanalytics_mobile_application_1"
+    snowplow__screen_context: "com_snowplowanalytics_mobile_screen_1"
+    snowplow__application_errors_events: "com_snowplowanalytics_snowplow_application_error_1"
```
[the remark this answers is not in the document]
Comment (agnessnowplow) [reply dbt_project.yml]: thanks!!

File: models/optional_modules/app_errors/scratch/redshift_postgres/snowplow_mobile_app_errors_this_run.sql [hunk 3]
```diff
@@ -104,17 +87,14 @@ select
...
-  ae.thread_name    
+  abe.message,
+  abe.programming_language,
+  abe.class_name,
+  abe.exception_name,
+  abe.is_fatal,
+  abe.line_number,
+  abe.stack_trace,
+  abe.thread_id,
+  abe.thread_name
```
Comment (rlh1994) [inline models/optional_modules/app_errors/scratch/redshift_postgres/snowplow_mobile_app_errors_this_run.sql:98]: These are all prefixed with `app_err_con` I think?

File: models/screen_views/scratch/redshift_postgres/snowplow_mobile_screen_views_this_run.sql
Comment (rlh1994) [inline models/screen_views/scratch/redshift_postgres/snowplow_mobile_screen_views_this_run.sql:1]: If this comment is here twice ignore the duplicate, but could we not just have one screen views this run file now, they are basically the same?

File: models/sessions/scratch/snowplow_mobile_sessions_aggs.sql [hunk 4]
```diff
@@ -13,6 +13,7 @@
 
 with events as (
   select
+    es.session_identifier,
```
Comment (rlh1994) [inline models/sessions/scratch/snowplow_mobile_sessions_aggs.sql:16]: What's the reason for adding this in here? We overwrite the session_id field anyway and it's in the main session table anyway?

File: models/sessions/scratch/snowplow_mobile_sessions_aggs.sql [hunk 5]
```diff
@@ -45,33 +47,32 @@ with events as (
...
-    group by 1
+    group by 1,2
   {% else %}
     select
-      {% if target.type == 'postgres' %}
-        cast(null as uuid) as session_id,
-      {% else %}
-        cast(null as {{type_string() }}) as session_id,
-      {% endif %}
+      cast(null as {{type_string() }}) as session_identifier,
```
Comment (rlh1994) [inline models/sessions/scratch/snowplow_mobile_sessions_aggs.sql:66]: Regardless of removing/keeping identifier in this table, keep this change because it might not be a uuid if they change it

File: models/sessions/scratch/snowplow_mobile_sessions_sv_details.sql [hunk 6]
```diff
@@ -8,6 +8,7 @@
 }}
 
 select
+  sv.session_identifier,
```
Comment (rlh1994) [inline models/sessions/scratch/snowplow_mobile_sessions_sv_details.sql:11]: Again don't think there is a real need for this to be added

**Review by rlh1994 — COMMENTED** [review]
I've got to review the integration tests stuff still, but this should give you something to keep you busy :)

File: models/screen_views/scratch/redshift_postgres/snowplow_mobile_screen_views_this_run.sql
[the remark this answers is not in the document]
Comment (emielver) [reply models/screen_views/scratch/redshift_postgres/snowplow_mobile_screen_views_this_run.sql:1]: Yeah good point, have combined

File: integration_tests/.scripts/integration_test_w_custom_module.sh
Comment (rlh1994) [inline integration_tests/.scripts/integration_test_w_custom_module.sh:1]: Do we ever even run this?

File: models/users/scratch/snowplow_mobile_users_lasts.sql [hunk 7]
```diff
@@ -1,20 +1,31 @@
...
   config(
     sort='device_user_id',
     dist='device_user_id',
     sql_header=snowplow_utils.set_query_tag(var('snowplow__query_tag', 'snowplow_dbt'))
-  ) 
+  )
 }}
 
 select
+  a.user_identifier,
```
Comment (rlh1994) [inline models/users/scratch/snowplow_mobile_users_lasts.sql:10]: Again, technically no reason for this to be here.

File: models/users/scratch/snowplow_mobile_users_aggs.sql [hunk 8]
```diff
@@ -12,14 +12,15 @@
 }}
 
 select
+  user_identifier,
```
Comment (rlh1994) [inline models/users/scratch/snowplow_mobile_users_aggs.sql:15]: Again, technically no reason for this to be here.

**Review by rlh1994 — CHANGES_REQUESTED** [review]
Last few comments

**Review by rlh1994 — COMMENTED** [review]
Last few comments

File: integration_tests/.scripts/integration_test_w_custom_module.sh
Comment (emielver) [reply integration_tests/.scripts/integration_test_w_custom_module.sh:1]: Nope, can remove

File: models/base/scratch/redshift_postgres/snowplow_mobile_base_events_this_run.sql
[the remark this answers is not in the document]
Comment (emielver) [reply models/base/scratch/redshift_postgres/snowplow_mobile_base_events_this_run.sql:1]: I'm not sure, it was last touched by Rahul 2 years ago so have deleted for now and we'll see!

**Review by rlh1994 — APPROVED** [review]
It would be good to test the app errors if possible before release, but otherwise it looks good to me!

**Comment by rlh1994** [comment]
image
Something is weird in the runs

File: dbt_project.yml [hunk 9]
```diff
@@ -29,20 +29,26 @@ vars:
...
+    snowplow__screen_view_events: com_snowplowanalytics_mobile_screen_view_1
     snowplow__sessions_table: "{{ ref('snowplow_mobile_sessions') }}" # Change to your custom sessions table if you have disabled the standard sessions table in favour of a custom version. Advanced config.
     # Variables - Standard Config
     snowplow__start_date: "2020-01-01"
     snowplow__backfill_limit_days: 30
     snowplow__app_id: []
     snowplow__platform: ['mob']
     snowplow__session_stitching: true
+    snowplow__user_stitching_id: 'user_id'
+    snowplow__session_identifiers: [{'schema' : 'com_snowplowanalytics_snowplow_client_session_1', 'field' : 'session_id', 'prefix': 'sc'}]
```
Comment (agnessnowplow) [inline dbt_project.yml:47]: I think this will not work properly for Bigquery users out-of-the box as it doesn't combine versions, perhaps it is better to make it target based (_*).

Comment (agnessnowplow) [reply dbt_project.yml:47]: Same with the user identifier.

Comment (rlh1994) [reply dbt_project.yml:47]: Don't we need to change that in unified then 👀

Comment (agnessnowplow) [reply dbt_project.yml:47]: We already did it, that was the first bug fix straight after release.

Comment (rlh1994) [reply dbt_project.yml:47]: https://github.com/snowplow/dbt-snowplow-unified/blob/d7365445f5a7fd345dc083881d6485ad9439cb7e/macros/identifiers.sql#L21 ?

Comment (rlh1994) [reply dbt_project.yml:47]: I didn't scroll, ignore me...

Comment (agnessnowplow) [reply dbt_project.yml:47]: https://github.com/snowplow/dbt-snowplow-unified/blob/d7365445f5a7fd345dc083881d6485ad9439cb7e/macros/identifiers.sql#L64

Comment (emielver) [reply dbt_project.yml:47]: Yeah for now the intended route is to list each of the columns separately. But this is functionality that would definitely be included in a future release!
